newrelic/resource_newrelic_nrql_drop_rule.go

```diff
@@ -0,0 +1,196 @@
+				Optional:    true,
+				Description: "Provides additional information about the rule.",
+			},
+			"rule_id": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The id, uniquely identifying the rule.",
+			},
+		},
+	}
+}
+
+func resourceNewRelicNRQLDropRuleCreate(d *schema.ResourceData, meta interface{}) error {
+	providerConfig := meta.(*ProviderConfig)
+
+	if !providerConfig.hasNerdGraphCredentials() {
+		return errors.New("err: NerdGraph support not present, but required for Create")
+	}
+
+	client := providerConfig.NewClient
+
+	accountID := d.Get("account_id").(int)
+
+	createInput := []nrqldroprules.NRQLDropRulesCreateDropRuleInput{
+		{
+			Description: d.Get("description").(string),
+			Action:      nrqldroprules.NRQLDropRulesAction(strings.ToUpper(d.Get("action").(string))),
+			NRQL:        d.Get("nrql").(string),
+		},
+	}
+
+	created, err := client.Nrqldroprules.NRQLDropRulesCreate(accountID, createInput)
+	if err != nil {
+		return err
+	}
+
+	rule := created.Successes[0]
+
+	id := fmt.Sprintf("%d:%s", created.Successes[0].AccountID, rule.ID)
+
+	d.SetId(id)
+
+	return resourceNewRelicNRQLDropRuleRead(d, meta)
+}
+
+func resourceNewRelicNRQLDropRuleRead(d *schema.ResourceData, meta interface{}) error {
+	providerConfig := meta.(*ProviderConfig)
+
+	if !providerConfig.hasNerdGraphCredentials() {
+		return errors.New("err: NerdGraph support not present, but required for Read")
+	}
+
+	client := providerConfig.NewClient
+
+	log.Printf("[INFO] Reading New Relic entity tags for entity guid %s", d.Id())
+
+	accountID, ruleID, err := getNRQLDropRuleIDs(d.Id())
+	if err != nil {
+		return err
+	}
+
+	rule, err := getNRQLDropRuleByID(client, accountID, ruleID)
+
+	if err != nil {
+		if _, ok := err.(*nrErrors.NotFound); ok {
+			d.SetId("")
+			return nil
+		}
+
+		return err
+	}
+
+	if err := d.Set("account_id", accountID); err != nil {
+		return err
+	}
+
+	if err := d.Set("rule_id", ruleID); err != nil {
+		return err
+	}
+
+	if err := d.Set("action", strings.ToLower(string(rule.Action))); err != nil {
+		return err
+	}
+
+	if err := d.Set("description", rule.Description); err != nil {
+		return err
+	}
+
+	if err := d.Set("nrql", rule.NRQL); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func resourceNewRelicNRQLDropRuleDelete(d *schema.ResourceData, meta interface{}) error {
+	providerConfig := meta.(*ProviderConfig)
+
+	if !providerConfig.hasNerdGraphCredentials() {
+		return errors.New("err: NerdGraph support not present, but required for Delete")
+	}
+
+	client := providerConfig.NewClient
+
+	log.Printf("[INFO] Deleting New Relic entity tags from entity guid %s", d.Id())
+
+	accountID, ruleID, err := getNRQLDropRuleIDs(d.Id())
+	if err != nil {
+		return err
+	}
+
+	deleteInput := []string{ruleID}
+
+	_, err = client.Nrqldroprules.NRQLDropRulesDelete(accountID, deleteInput)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func getNRQLDropRuleIDs(id string) (int, string, error) {
```

Comment:
`parse` is probably a better name here, I was confused above when looking at the code why there were two `get` calls for the read.